File: 70_Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] count = new int[2];
        // init
        count[0] = 1;   // distance 1: one way
        count[1] = 2;   // distance 2: two ways

        if (n <= 2) return count[n-1];

        for (int i = 3; i <= n; i++) {
            int tmp = count[0] + count[1];
            count[0] = count[1];
            count[1] = tmp;
        }

        return count[1];
    }
}

File: 70_Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1: return 1
        # 1. init: location_0 and location_1
        prev = cur = 1
        # 2. update
        for _ in range(2, n+1):
            prev, cur = cur, prev + cur
        return cur